### DISCUSSION TITLE
Project Quay vs redhat Quay

### DISCUSSION THREAD

**Discussion by dkoci** [quay/quay#938] [Q&A]
Hi.
What is the difference between project quay and rh quay? If I understand correctly, Project quay is community edition of quay. Does it mean it is free for use? Is rh subscription mandatory for project quay?

Thanks for answers

**Comment by BillDett** [comment]
Hi Davorin-

That's right- Red Hat Quay is the commercial product that requires a 
subscription.  Project Quay is our 'upstream' version that is free to 
use and community supported (also supported by the Red Hat Quay 
engineers as much as we can).  Please use it as you like and feel free 
to ask questions and contribute back to the project via Pull Requests!

Bill

On 10/20/21 5:24 AM, Davorin wrote:
>
> Hi.
> What is the difference between project quay and rh quay? If I 
> understand correctly, Project quay is community edition of quay. Does 
> it mean it is free for use? Is rh subscription mandatory for project quay?
>
> Thanks for answers
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/quay/quay/discussions/938>, or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AABQYQVWFDWS3QWJFA5XPRTUH2DDPANCNFSM5GLES6FA>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>
>
-- 
Bill Dettelback
Senior Engineering Manager - Red Hat Quay, OpenShift Image Registry
M: 732-991-0052
https://www.redhat.com/en/technologies/cloud-computing/quay
Red Hat respects your work life balance. Therefore there is no need to answer this email out of your office hours.

**Reply by dkoci** [reply]
'Deploy Project Quay - High Availability' demo uses Ceph as an object storage. Is Ceph free for use? It is open-source, but links in quay documentation point to redhat sites. Are there different versions of Ceph (free one and paid one by redhat)?

tnx

**Reply by BillDett** [reply]
Red Hat produces enterprise software from open source projects- you should be able to use the FOSS Ceph version with Quay just fine but we QE test against Red Hat's version, so there may be some differences in set up or configuration.  Quay really just wants an S3 compatible endpoint for its object storage, so if you are looking for a free registry solution you have lots of other options (e.g. minio).

**Reply by dkoci** [reply]
Thank you for your answers. We have private s3 storage in our company.
Just one question. Is it possible to use openshift operator for project quay? There is only redhat quay operator in operator hub on openshift.

**Reply by flavianmissi** [reply]
You can build a catalog yourself from https://github.com/quay/quay-operator. In the future we plan to publish our operators to [https://operatorhub.io/](https://operatorhub.io/?keyword=quay), but ATM the versions there are not up-to-date so you're better off building and adding the catalog to your cluster yourself, as instructed on the quay operator repository README.

**Reply by dkoci** [reply]
This thing does not work. Operator is successfully installed, but registry resource not. I also tried redhat quay operator which is working like a charm.

**Reply by flavianmissi** [reply]
Hey @dkoci, could you tell us a little more about how you installed the operator, and share the operator logs from the failed attempt to deploy the registry?

**Reply by dkoci** [reply]
`I1210 12:47:05.558299       1 request.go:655] Throttling request took 1.032739768s, request: GET:https://172.30.0.1:443/apis/node.k8s.io/v1?timeout=32s
2021-12-10T12:47:07.471Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":8080"}
2021-12-10T12:47:07.472Z	INFO	controller-runtime.injectors-warning	Injectors are deprecated, and will be removed in v0.10.x
2021-12-10T12:47:07.472Z	INFO	controller-runtime.injectors-warning	Injectors are deprecated, and will be removed in v0.10.x
2021-12-10T12:47:07.472Z	INFO	controller-runtime.injectors-warning	Injectors are deprecated, and will be removed in v0.10.x
2021-12-10T12:47:07.472Z	INFO	controller-runtime.injectors-warning	Injectors are deprecated, and will be removed in v0.10.x
2021-12-10T12:47:07.472Z	INFO	setup	starting server on port 7071
2021-12-10T12:47:07.472Z	INFO	setup	starting manager
2021-12-10T12:47:07.472Z	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}
2021-12-10T12:47:07.472Z	INFO	controller-runtime.manager.controller.quayregistry	Starting EventSource	{"reconciler group": "quay.redhat.com", "reconciler kind": "QuayRegistry", "source": "kind source: /, Kind="}
2021-12-10T12:47:07.473Z	INFO	controller-runtime.manager.controller.quayecosystem	Starting EventSource	{"reconciler group": "redhatcop.redhat.io", "reconciler kind": "QuayEcosystem", "source": "kind source: /, Kind="}
2021-12-10T12:47:07.574Z	INFO	controller-runtime.manager.controller.quayecosystem	Starting Controller	{"reconciler group": "redhatcop.redhat.io", "reconciler kind": "QuayEcosystem"}
2021-12-10T12:47:07.574Z	INFO	controller-runtime.manager.controller.quayregistry	Starting Controller	{"reconciler group": "quay.redhat.com", "reconciler kind": "QuayRegistry"}
2021-12-10T12:47:07.574Z	INFO	controller-runtime.manager.controller.quayecosystem	Starting workers	{"reconciler group": "redhatcop.redhat.io", "reconciler kind": "QuayEcosystem", "worker count": 1}
2021-12-10T12:47:07.574Z	INFO	controller-runtime.manager.controller.quayregistry	Starting workers	{"reconciler group": "quay.redhat.com", "reconciler kind": "QuayRegistry", "worker count": 1}
2021-12-10T12:47:07.574Z	INFO	controllers.QuayRegistry	begin reconcile	{"quayregistry": "quay-ent/example-registry"}
2021-12-10T12:47:07.675Z	INFO	controllers.QuayRegistry	successfully retrieved referenced `configBundleSecret`	{"quayregistry": "quay-ent/example-registry", "configBundleSecret": "init-config-bundle-secret", "resourceVersion": "94235858"}
2021-12-10T12:47:07.703Z	INFO	controllers.QuayRegistry	cluster supports `Routes` API
2021-12-10T12:47:08.306Z	INFO	controllers.QuayRegistry	detected router canonical hostname: apps.okd-test.ts.telekom.si
2021-12-10T12:47:08.445Z	INFO	controllers.QuayRegistry	cluster supports `ObjectBucketClaims` API
2021-12-10T12:47:08.445Z	INFO	controllers.QuayRegistry	`ObjectBucketClaim` not found
2021-12-10T12:47:08.445Z	INFO	controllers.QuayRegistry	monitoring is only supported in AllNamespaces mode. Disabling component monitoring
E1210 12:47:08.446071       1 runtime.go:78] Observed a panic: "unknown component: tls" (unknown component: tls)
goroutine 436 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1db18a0, 0xc00054fc90)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x86
panic(0x1db18a0, 0xc00054fc90)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/quay/quay-operator/pkg/kustomize.ContainsComponentConfig(0xc0006b1590, 0xc00086ee7a, 0x3, 0x20f9300, 0x0, 0x0, 0x0)
	/workspace/pkg/kustomize/secrets.go:234 +0x818
github.com/quay/quay-operator/controllers/quay.(*QuayRegistryReconciler).Reconcile(0xc000622b40, 0x24559c8, 0xc000844a50, 0xc00086ed80, 0x8, 0xc00086ed60, 0x10, 0xc000844a50, 0xc00003a000, 0x1f50520, ...)
	/workspace/controllers/quay/quayregistry_controller.go:236 +0x1c48
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000852fa0, 0x2455920, 0xc0009aa240, 0x1eec280, 0xc0001e7240)
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:297 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000852fa0, 0x2455920, 0xc0009aa240, 0x2000000000000)
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:252 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x2455920, 0xc0009aa240)
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000beef50)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00061ff50, 0x2418160, 0xc0008449c0, 0xc0009aa201, 0xc0007b4120)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000beef50, 0x3b9aca00, 0x0, 0xc000bd6e01, 0xc0007b4120)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x2455920, 0xc0009aa240, 0xc0007b3810, 0x3b9aca00, 0x0, 0x1)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x2455920, 0xc0009aa240, 0xc0007b3810, 0x3b9aca00)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:212 +0x40d
panic: v1.ComponentKind("unknown component: tls") [recovered]
	panic: v1.ComponentKind("unknown component: tls")

goroutine 436 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x109
panic(0x1db18a0, 0xc00054fc90)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/quay/quay-operator/pkg/kustomize.ContainsComponentConfig(0xc0006b1590, 0xc00086ee7a, 0x3, 0x20f9300, 0x0, 0x0, 0x0)
	/workspace/pkg/kustomize/secrets.go:234 +0x818
github.com/quay/quay-operator/controllers/quay.(*QuayRegistryReconciler).Reconcile(0xc000622b40, 0x24559c8, 0xc000844a50, 0xc00086ed80, 0x8, 0xc00086ed60, 0x10, 0xc000844a50, 0xc00003a000, 0x1f50520, ...)
	/workspace/controllers/quay/quayregistry_controller.go:236 +0x1c48
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000852fa0, 0x2455920, 0xc0009aa240, 0x1eec280, 0xc0001e7240)
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:297 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000852fa0, 0x2455920, 0xc0009aa240, 0x2000000000000)
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:252 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x2455920, 0xc0009aa240)
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000beef50)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00061ff50, 0x2418160, 0xc0008449c0, 0xc0009aa201, 0xc0007b4120)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000beef50, 0x3b9aca00, 0x0, 0xc000bd6e01, 0xc0007b4120)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x2455920, 0xc0009aa240, 0xc0007b3810, 0x3b9aca00, 0x0, 0x1)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x2455920, 0xc0009aa240, 0xc0007b3810, 0x3b9aca00)
	/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:212 +0x40d
`

**Reply by flavianmissi** [reply]
That's strange, what version of the operator have you deployed to your cluster? And what does your `QuayRegistry` CR look like?
Our CI uses a [deploy script](https://github.com/quay/quay-operator/blob/master/hack/deploy.sh#L2-L26) to install quay via operator in an OCP cluster, and it works fine there. You might want to try that. The scripts are fairly new, so we haven't gotten around documenting the flow in the README yet but there's enough docs in the shell files themselves to get you going.
Let us know how it goes.

**Reply by dkoci** [reply]
I have deplooyed operator from master branch (used README.md as a reference).
Registry CR (also created config-bundle-secret):

`spec:
  configBundleSecret: config-bundle-secret
  components:
    - managed: false
      kind: clair
    - managed: true
      kind: postgres
    - managed: false
      kind: objectstorage
    - managed: true
      kind: redis
    - managed: false
      kind: horizontalpodautoscaler
    - managed: true
      kind: route
    - managed: false
      kind: mirror
    - managed: false
      kind: monitoring
    - managed: true
      kind: tls`

**Reply by dkoci** [reply]
> That's strange, what version of the operator have you deployed to your cluster? And what does your `QuayRegistry` CR look like? Our CI uses a [deploy script](https://github.com/quay/quay-operator/blob/master/hack/deploy.sh#L2-L26) to install quay via operator in an OCP cluster, and it works fine there. You might want to try that. The scripts are fairly new, so we haven't gotten around documenting the flow in the README yet but there's enough docs in the shell files themselves to get you going. Let us know how it goes.

There is redhat-quay operator image in that script instead of project-quay operator. Is that correct? I am confused, beacause as I know, redhat-quay is not for free. Am I correct?

**Reply by dkoci** [reply]
Which is the latest stable version, which would be suitable for production?

**Reply by flavianmissi** [reply]
you should be good with 3.6.2

**Reply by dkoci** [reply]
I deployed it from deploy script as you suggested. Is that redhat quay operator which needs subscription?
What about channel. Is test channel ok? I am also confused about 3.6.2 version, because it is unstable?

![image](https://user-images.githubusercontent.com/5565897/146231597-238cdb2b-1d32-4e6a-add5-e75a43a966c0.png)

**Reply by flavianmissi** [reply]
I'm glad it worked with the script @dkoci! the "unstable" in the tag is because we build it with whatever is on the `redhat-3.6` branch, right now what's there is exactly the same as what's on quay 3.6.2, but as we move on to the next 3.6 version you may see things that aren't fully tested ending up on the 3.6-unstable tag.
We'll eventually have a better packed operator (via operatorhub.io) so it will feel more like the real quay rather than a test build for Red Hat Quay (that's way you see test in the names), but for now you're good to go with that install.

**Reply by dkoci** [reply]
What about updates? Does the operator take care of it or is it a manual task?
Btw it is really great product, I am impressed. Missing few things (e.g. prevent users without superuser access from creating organizations; enabling user creation in case of LDAP authentication; and a bit more work on the documentation - it is a bit confusing for new quay users). But great job anyway :)

**Reply by flavianmissi** [reply]
In the future when using the operatorhub.io catalog you'll get updates we add them to the catalog, but for now you'll have to build new catalogs yourself (the script will do this for you).
Thanks! Yeah I get that it's confusing for new users, we're slowly improving these things as we moved forward - your feedback is appreciated! 
Please also feel free to send us patches, getting documentation PRs in should go fairly easy.
